cpp/arcticdb/python/python_to_tensor_frame.cpp

```diff
@@ -68,19 +68,34 @@ NativeTensor obj_to_tensor(PyObject *ptr) {
             auto none = py::none{};
             auto obj = reinterpret_cast<PyObject **>(arr->data);
             bool empty = false;
+            bool all_nans = false;
             PyObject *sample = *obj;
             if (sample == none.ptr() || is_py_nan(sample)) {
-                // Iterate till we find the first non null element, the frontend ensures there is at least one.
+                empty = true;
+                all_nans = true;
                 util::check(c_style, "Non contiguous columns with first element as None not supported yet.");
-                auto casted_col =
-                    std::find_if(obj, obj + size, [&none](auto val) { return val != none.ptr() && !is_py_nan(val); });
-                empty = casted_col == obj + size;
-                sample = *casted_col;
+                PyObject** current_object = obj;
+                while(current_object < obj + size) {
+                    if(*current_object == none.ptr()) {
+                        all_nans = false;
+                    } else if(is_py_nan(*current_object)) {
+                        empty = false;
+                    } else {
+                        all_nans = false;
```

Comment:
```suggestion
                        // `current_object` is an usual scalar value.
                        all_nans = false;
```

Reply:
I've added this explanation as a comment in the code